gomock/call.go

```diff
@@ -149,6 +150,27 @@ func (c *Call) Do(f interface{}) *Call {
 				vargs[i] = reflect.Zero(ft.In(i))
 			}
 		}
+		defer func() {
+			if r := recover(); r != nil {
+				errMsg, ok := r.(string)
+
+				// We only handle a very specific panic
```

Comment:
Do you see a use case where a user might want to panic instead of failing the test?

If not, WDYTA letting `fmt.Sprint` do some work here (https://play.golang.org/p/SJh50xjMu_8) and allow more than just strings?